src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -636,13 +636,48 @@ static void TransformationUpToCPUSpecificOpSet(std::shared_ptr<ngraph::Function>
 
     // Execute before snippets. Otherwise FQ will be converted to Subgraph
     postLPTPassManager.register_pass<ConvertFqRnnToQuantizedRnn>();
+
+    // Float MHA is supported by snippets now
+    auto postLPTPassConfig = postLPTPassManager.get_pass_config();
+    if (!_enableBF16) {
+        postLPTPassConfig->disable<MHAFloatFusion>();
+        postLPTPassConfig->disable<MHAFloatFusion2>();
+    }
+
     postLPTPassManager.run_passes(nGraphFunc);
 
     if (_snippetsMode != Config::SnippetsMode::Disable && dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx2)) {
         ngraph::pass::Manager snippetsManager;
         if (_snippetsMode != Config::SnippetsMode::IgnoreCallback)
             snippetsManager.register_pass<SnippetsMarkSkipped>();
         snippetsManager.register_pass<ngraph::snippets::pass::EnumerateNodes>();
+        if (!_enableBF16) {
+            // TODO: Need to add BF16 support for MHA in Snippets
+            snippetsManager.register_pass<ngraph::snippets::pass::TokenizeMHASnippets>();
+            snippetsManager.get_pass_config()->set_callback<ngraph::snippets::pass::TokenizeMHASnippets>(
```

Comment:
I think is it makes sense to encapsulate TokenizeMHASnippets into general TokenizeSnippets pass? I would prefer to have minimal number of passes need to be directly called on plugin level.

Reply:
I have created the common pass Snippets Tokenization which includes common snippets pipeline: Enumeration, MHA Tokenization, common Tokenization and common Optimizations. Take a look one more time, please